### PR TITLE
Incorporate `dyn` into more comments and docs.

### PR DESCRIPTION
**Pull request by frewsxcv**
r? @rust-lang/docs

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
I'm not sure to see the why of this PR. It works without `dyn` so why adding it?

**Comment by steveklabnik** [comment]
> It works without dyn so why adding it?

`dyn` is considered idiomatic, so this is updating the code to be idiomatic like any other update.

**Comment by steveklabnik** [comment]
r=me after the conflict has been fixed

File: src/test/run-pass/string-box-error.rs [hunk 1]
```diff
@@ -8,7 +8,8 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-// Ensure that both `Box<Error + Send + Sync>` and `Box<Error>` can be obtained from `String`.
+// Ensure that both `Box<dyn Error + Send + Sync>` and `Box<dyn Error>` can be
+// obtained from `String`.
```
Comment (RalfJung) [inline src/test/run-pass/string-box-error.rs]: Shouldn't the types in the code below be consistent with the ones in the comment here?

**Pull request merged**